compiler/rustc_mir/src/transform/check_consts/validation.rs

```diff
@@ -282,6 +287,27 @@ impl Validator<'mir, 'tcx> {
         }
     }
 
+    fn local_has_storage_dead(&mut self, local: Local) -> bool {
+        let ccx = self.ccx;
+        self.local_has_storage_dead
+            .get_or_insert_with(|| {
+                struct StorageDeads {
+                    locals: BitSet<Local>,
+                }
+                impl Visitor<'tcx> for StorageDeads {
+                    fn visit_statement(&mut self, stmt: &Statement<'tcx>, _: Location) {
+                        if let StatementKind::StorageDead(l) = stmt.kind {
```

Comment:
So each time we check the table for a different local, we scan the entire MIR for all locals and then throw away the result except for this one local? How does this make any sense?

Reply:
The `BitSet` is computed once and memoized. We use this pattern for a few other such checks. If there are no borrows, the `BitSet` is never created.

Reply:
Oh, I got the brace nesting wrong... `Option` has a `get_or_insert_with` method?!?